src/components/List/tokenList.js

```diff
@@ -0,0 +1,40 @@
+import React, { useState } from 'react';
+import getToken from '../../lib/tokens';
+import { Router, Redirect, Route } from 'react-router';
+
+export const AddToken = () => {
+  const tokenLocal = localStorage.getItem('token');
+  const [token, setToken] = useState(tokenLocal);
+
+  const handleNewToken = () => {
+    if (!tokenLocal) {
+      const newToken = getToken();
+      localStorage.setItem('token', newToken);
+    }
+    window.location.href = '/list';
```

Comment:
👍 This works, it's not the "react router way" of doing things, but gets the job done - nice work @lizhtml and @AndrewEspejo

👨‍🏫 In future implementations, you should use React Router's `useHistory` hook to navigate between pages.

```javascript
// Taken from React Router docs: https://reactrouter.com/web/api/Hooks/usehistory
import { useHistory } from "react-router-dom";

function HomeButton() {
  let history = useHistory();

  function handleClick() {
    // This is what's taking care of the navigation.
    history.push("/home");
  }

  return (
    <button type="button" onClick={handleClick}>
      Go home
    </button>
  );
}
```

Reply:
Thanks for the documentation it is very valuable to continue in our learning process.

Reply:
overall, I like the architecture of tje project. Pretty nice job

Reply:
This is not necessary, but I think is a good thing to learn and to have in mind [Protected Components](https://medium.com/@subalerts/creating-protected-routes-in-react-js-89e95974a822)